src/Microsoft.AspNetCore.Http.Abstractions/PathString.cs

```diff
@@ -71,20 +72,45 @@ public string ToUriComponent()
             {
                 return string.Empty;
             }
-            var values = _value.Split(splitChar);
-            var changed = false;
-            for (var i = 0; i < values.Length; i++)
-            {
-                var value = values[i];
-                values[i] = UrlEncoder.Default.Encode(value);
 
-                if (!changed && value != values[i])
+            StringBuilder buffer = null;
```

Comment:
Who usually calls this method and how expensive is this change? Looks like we never measured it..